### PR TITLE
Add AssignCulture task back to CoreCLR build

### PR DESCRIPTION
**Pull request by AndyGerlicher**
CultureInfo.GetCultures isn't supported, so I just hard coded a small list. There are like 800 more, so this isn't great but it got it up and building quickly.

### CODE REVIEW

**Comment by akoeplinger** [comment]
FYI the aspnet team ran into this as well (see https://github.com/dotnet/corefx/issues/1669) and ended up with [CultureInfoList.cs](https://github.com/aspnet/Localization/blob/5e1fb16071affd15f15b9c732833f3ae2ac46e10/src/Microsoft.Framework.Globalization.CultureInfoCache/CultureInfoList.cs).

**Comment by AndyGerlicher** [comment]
Thanks! I'll use that instead.

**Comment by AndyGerlicher** [comment]
@dsplaisted @ValMenn @rainersigwald Can one of you take a look at this please? Most of it is just copied from aspnet and replaced the old array + binary search method. I did modify it to use CultureInfo.GetCultures if it was available though.

File: src/XMakeTasks/AssignCulture.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ namespace Microsoft.Build.Tasks
     /// 
     ///      MyGlyph.fr.bmp          ==> MyGlyph.bmp [Culture='fr']
     /// 
-    /// This is because embedded resources are refered to this way.
+    /// This is because embedded resources are referred to this way.
```
Comment (rainersigwald) [inline src/XMakeTasks/AssignCulture.cs]: :heart:

File: src/XMakeTasks/CultureInfoCache.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace Microsoft.Framework.Globalization
+{
+    /// <summary>
+    /// Provides read-only cached instances of <see cref="CultureInfo"/>.
+    /// <remarks>
+    /// Source: https://raw.githubusercontent.com/aspnet/Localization/dev/src/Microsoft.Framework.Globalization.CultureInfoCache/CultureInfoCache.cs
```
Comment (rainersigwald) [inline src/XMakeTasks/CultureInfoCache.cs]: Can you specify the commit hash you're pulling from in the commit message?

(I'll spare you my super-nitpicky "bring this in by itself via `--orphan`, then merge it in" nonsense :smile:)

**Comment by rainersigwald** [comment]
LGTM

**Comment by dsplaisted** [comment]
Based on the discussion in dotnet/corefx#1669, I'm not sure hard-coding the list of supported cultures is appropriate here.  It seems like perhaps what we should be doing instead to check if a culture string is valid is either to try creating a CultureInfo and seeing if that throws, or to check if it matches the language tag syntax defined in [BCP 47](http://tools.ietf.org/html/bcp47) Section 2.1.

**Comment by AndyGerlicher** [comment]
Ok. For non CoreCLR it stays a similar implementation (calls GetCultures to populate that cache). For CoreCLR it will new up the CultureInfo when asked to check if it's valid.

File: src/XMakeTasks/CultureInfoCache.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        /// Populate the cache with CultureInfo.GetCulture (if supported).
+        /// </summary>
+        private static void PopulateCultures()
+        {
+            if (s_cache.Count > 0) return;
+
+#if FEATURE_CULTUREINFO_GETCULTURES
+            var culturesToAdd = CultureInfo.GetCultures(CultureTypes.AllCultures);
+#else
+            // In CoreCLR we'll add at least the current thread to the cache, but populate the list as they're requested.
```
Comment (rainersigwald) [inline src/XMakeTasks/CultureInfoCache.cs]: Thread?

**Comment by rainersigwald** [comment]
:shipit:

**Pull request merged**